www/js/Widget.js

```diff
@@ -215,11 +215,18 @@ class Widget {
 
   on (eve, callback) {
     if (!this.events[eve]) {
-      return console.error(`Widget: ${eve} is not a Widget event`)
+      // return console.error(`Widget: ${eve} is not a Widget event`)
+      this.events[eve] = []
     }
     this.events[eve].push(callback)
   }
 
+  emit (eve, e) {
+    this.events[eve].forEach(e => {
+      e(this.events)
```

Comment:
the second arg in emit is the event object (the data the user passes in) so for example:
```js
this.emit('new-position', { x: 0, y: 10, z: 0})
```
so that's what needs to get passed into the callback functions when u call them (rather than passing in the list of events like u are now), problem here is that u're using the same variable name "e" for the callback functions, essentially overriding the second arg being passed in (which is also called "e")